### ISSUE TITLE
fcntl.open with two arguments is missing in 0.4.0-M2

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#1613]
I dusted off some two month code that was working in 0.4.0-SNAPSHOT pre M1 to
send to a colleague.  The code uses the quite commonly used idiom
of fcntl.open with two arguments.  

It looks like the recent (2121d12f6bccc4dbd23c4d9d676b1cadfcb27c6c) varargs change to fcntl causes the two argument form  to no longer work.     

I looked tor a/an *fcnt*lSuite to verify what I thought I was seeing and did not
find anything.  Mea culpa if I missed something obvious.

I believe, but have not tried out. That a/the solution is to declare the two 
argument form in fcntl.scala, That is, without the mode.
 
If I can replicate this, say by creating an fcntlSuite.scala, I will PR soonest.

**Comment by densh** [comment]
Thanks for the report! Please attach a failing test here as an issue comment.

**Comment by LeeTibbert** [comment]
My thanks to @david-bouyssie for the discussions which lead to the discovery and fix for
this Issue.

I was able to replicate, so I submitted PR #1614.  The commit message for that PR
contains the original failure message. The new FcntlSuite.scala contains a unit-test
which fails on 0.3.4-M2 and passes after the PR.

I can post both here is that makes tracking Issues easier.

**Comment by LeeTibbert** [comment]
In the hopes of making life easier for others following along, I mention the workaround
for 0.4.0-M2: the three argument form of fcntl.open(). Not an astonishing discovery,
but may save someone some futzing around. I use S_IRUSR because almost
all file owners have read permission to a file they are opening.  Other 
mode_t values may work as well.

```
 val fd = fcntl.open(toCString(fileName), fcntl.O_RDWR, stat.S_IRUSR)
```

**Comment by sjrd** [comment]
Fixed in b3bad595f53fb5651a8d732c39f366ca67c4eb15.
